### PR TITLE
Export (filtered) cancer variants

### LINKED ISSUE #1914: [Feature request] Option to show number of filtered variants and export filtered variants table
After applying filters to variants, there is no total count. It is hard to say even if filtered removed anything. It'd be great to show a total count of filtered variants. And then export them in excel format. There is option to export causative variants on institute level but not on case level. Of course one can pin these variants one by one and print out a report, but exporting before pinning can be quite useful.

### PR DESCRIPTION
**Pull request by northwestwitch**
Fix #1914

Adds a button "Filter and export" to cancer variants page. When pressed the filtered variants are exported to CSV. 

This is how the page is now:
image

This is how the page becomes:
image

**How to test**:
1. Install the branch on Scout stage and go to a cancer case, for instance: https://scout-stage.scilifelab.se/cust000/strongfly
1. Test filtering and export of cancer variants

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by
- [ ] tests executed by

### CODE REVIEW

File: scout/server/blueprints/variants/controllers.py [hunk 1]
```diff
@@ -416,6 +394,43 @@ def variant_export_lines(store, case_obj, variants_query):
     return export_variants
 
 
+def variant_export_genes_info(store, gene_list):
```
Comment (northwestwitch) [inline scout/server/blueprints/variants/controllers.py:397]: Moved this code to another function to simplify the variant_export_lines function.

**Comment by northwestwitch** [comment]
Ready for review.

File: scout/server/blueprints/variants/controllers.py [hunk 2]
```diff
@@ -416,6 +394,43 @@ def variant_export_lines(store, case_obj, variants_query):
...
+        gene_name = gene(store, hgnc_id)["symbol"]
+
+        gene_ids.append(hgnc_id)
+        gene_names.append(gene_name)
+
+        hgvs_nucleotide = "-"
+        # gather HGVS info from gene transcripts
+        transcripts_list = gene_obj.get("transcripts")
+        for transcript_obj in transcripts_list:
+            if transcript_obj.get("is_canonical") and transcript_obj.get("is_canonical") is True:
```
[the remark this answers is not in the document]
Comment (northwestwitch) [reply scout/server/blueprints/variants/controllers.py:423]: Better to be very sure? 😆

**Comment by northwestwitch** [comment]
> Exporting all Clinical Structural variants creates an inconsistent csv-file. Checked with https://csvlint.io. I was running default demo cancer. Exported all.
> 
> Rows 6, 12, 13 and 16 have 14 columns. All other rows have 13. Example:
> 
> `0.0,1,2752092,A>[19:48324960[CA,2752092_A>[19:48324960[CA,-,-,-,-,121,17,-1,138,0,-1`
> 
> [cancer_case-filtered_sv-variants-3.csv.txt](https://github.com/Clinical-Genomics/scout/files/4769790/cancer_case-filtered_sv-variants-3.csv.txt)

Thanks, I'll check!

File: scout/server/blueprints/variants/controllers.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (northwestwitch) [reply scout/server/blueprints/variants/controllers.py:423]: I'll improve the code here but it should be a double check here: first if the is_canonical key exists and then if it has a boolean value that is True

**Comment by northwestwitch** [comment]
Fixed according to comments @mikaell

File: scout/server/blueprints/variants/controllers.py [hunk 3]
```diff
@@ -374,9 +374,11 @@ def variant_export_lines(store, case_obj, variants_query):
             variant_line += gene_info
         else:
             i = 0
-            while i < 4:
-                variant_line.append("-")  # instead of gene ids
-                i = i + 1
+            while i < 3:
```
Comment (mikaell) [inline scout/server/blueprints/variants/controllers.py:377]: Unnamed constant in while-loop (4).

Comment (northwestwitch) [reply scout/server/blueprints/variants/controllers.py:377]: It's a good suggestion, I'll rename it to empty_col, thanks!

**Comment by northwestwitch** [comment]
> 1. Linebreaks are '\n' and not '\r\n'. See [RFC](https://tools.ietf.org/html/rfc4180). Maybe no-one will notice.

I don't think I'm adding any \r in the exported text file, see code:

image

Could it be that the \r is added by the editor or the program you used to open  it?

**Comment by northwestwitch** [comment]
@mikaell I've fixed the variant name to something more self-explanatory. I don't think the code is responsible to add any \r at the end of the lines, just \n.

**Comment by mikaell** [comment]
> > 1. Linebreaks are '\n' and not '\r\n'. See [RFC](https://tools.ietf.org/html/rfc4180). Maybe no-one will notice.
> 
> I don't think I'm adding any \r in the exported text file, see code:
> 
> image
> 
> Could it be that the \r is added by the editor or the program you used to open it?

I meant '\r' is missing if we want to strictly follow the RFC:

>       Each record is located on a separate line, delimited by a line
>       break (CRLF).  For example:
>
>       aaa,bbb,ccc CRLF
>       zzz,yyy,xxx CRLF

This is no big deal, most editors and other software handles it just fine.

**Comment by northwestwitch** [comment]
> I meant '\r' is missing if we want to strictly follow the RFC:

Ah no, we don't want to follow the RFC! These \r always cause a lot of trouble when parsing files, I'd gladly avoid them!

**Comment by northwestwitch** [comment]
@mikaell if the PR is fine could you approve?

**Comment by northwestwitch** [comment]
Thanks @mikaell !

**Pull request merged**